test/suite/test_compat05.py

```diff
@@ -0,0 +1,85 @@
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_compat05.py
+# Check log.archive compatibility API
+
+import fnmatch, os, time
+import wiredtiger, wttest
+from suite_subprocess import suite_subprocess
+from wtdataset import SimpleDataSet, simple_key
+from wtscenario import make_scenarios
+
+class test_compat05(wttest.WiredTigerTestCase, suite_subprocess):
+    remove_values = [
+        ('archive-false', dict(remove_arg = 'archive=false', removed = False)),
+        ('archive-true', dict(remove_arg = 'archive=true', removed = True)),
+        ('default', dict(remove_arg = '', removed = True)),
+        ('remove-false', dict(remove_arg = 'remove=false', removed = False)),
+        ('remove-override-order1', dict(remove_arg = 'archive=false,remove=true', removed = True)),
+        ('remove-override-order2', dict(remove_arg = 'remove=true,archive=false', removed = True)),
+        ('remove-true', dict(remove_arg = 'remove=true', removed = True)),
+    ]
+    scenarios = make_scenarios(remove_values)
+
+    log = 'WiredTigerLog.0000000001'
+
+    # Create the database with logging configured.
+    def conn_config(self):
+        return 'create,log=(enabled,file_max=100K,' + self.remove_arg + ')'
+
+    # Check if the log file has been removed.
+    def check_remove(self):
+        removed = False
+        for i in range(1,90):
+            # Sleep and then see if log removal ran. We do this in a loop
+            # for slow machines. Max out at 90 seconds.
+            time.sleep(1.0)
+            if not os.path.exists(self.log):
+                removed = True
+                break
+        return removed
+
+    # Run a single test.
+    def test_compat05(self):
+        # Populate the database to create some log files.
+        uri = 'table:test_compat05'
+        ds = SimpleDataSet(self, uri, 10000, key_format='S', value_format='S')
+        ds.populate()
+
+        # Assert there's a first log file.
+        self.assertTrue(os.path.exists(self.log))
```

Comment:
Do you want to assert there is a second log too? If there is only one log file and then you checkpoint, in log file 1, it cannot remove the log file with the checkpoint in it. So you want to assert there's a 2nd log file where the checkpoint will go, so that log file 1 can get removed.

Reply:
Fixed in 357c7585e.